src/server/journal/serializer.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <optional>
+#include <string>
+
+#include "base/io_buf.h"
+#include "io/io.h"
+#include "server/common.h"
+#include "server/journal/types.h"
+#include "server/main_service.h"
+
+namespace dfly {
+
+// JournalWriter serializes journal entries to a sink.
+// It automatically keeps track of the current database index.
+struct JournalWriter {
+  // Initialize with sink and optional start database index. If no start index is set,
+  // a SELECT will be issued before the first entry.
+  JournalWriter(io::Sink* sink, std::optional<DbIndex> dbid = std::nullopt);
```

Comment:
do you use the dbid in any flow? I didnt see this flow in the bigger PR.
If you do initialize with db id than caller mast make sure to call the select this is a bit confusing

Reply:
I do. In which case must the caller care about the id? The should absorb all select entries